kingpin/actors/rightscale/test/test_api.py

```diff
@@ -206,10 +206,50 @@ def test_launch_server_array(self):
 
         ret = yield self.client.launch_server_array(array_mock)
         self.mock_client.server_arrays.launch.assert_called_once_with(
-            res_id=1234)
+            res_id=1234, params=None)
         self.assertEquals(ret, instance_mock)
 
     @testing.gen_test
+    def test_launch_server_array_launch_0_instance(self):
+        array_mock = mock.MagicMock(name='fake_array')
+        array_mock.soul = {'name': 'fake array to launch'}
+        array_mock.self.path = '/a/b/1234'
+
+        # A count of 1 should pass params=None to the launch call
+        ret = yield self.client.launch_server_array(array_mock, count=0)
+        self.assertEquals(ret, None)
+
+        # A count of 1 should pass params=None to the launch call
+        ret = yield self.client.launch_server_array(array_mock, count=None)
+        self.assertEquals(ret, None)
```

Comment:
I think that the more important thing to test here is that `self._client.server_arrays.launch` is not called, rather than that the return value of `launch_server_array` is None.